### ISSUE TITLE
Define resolver for Custom scalar type

### ISSUE THREAD

**Issue by valburyakov** [nestjs/graphql#9]
I'm trying to migrate my old nestjs app with graphQL to new GraphqlModule. And i faced with problem how to define resolver for scalar type, previously i had:
```
scalar Date

type User {
  _id: ID!
  fullName: String
  email: String
  hireDate: Date
  pictureUrl: String
}
```
```
import { GraphQLScalarType } from 'graphql';
import { Kind } from 'graphql/language';

export const resolver = {
  Date: new GraphQLScalarType({
    name: 'Date',
    description: 'Date custom scalar type',
    parseValue(value) {
      return new Date(value); // value from the client
    },
    serialize(value) {
      return value.getTime(); // value sent to the client
    },
    parseLiteral(ast) {
      if (ast.kind === Kind.INT) {
        return parseInt(ast.value, 10); // ast value is always in string format
      }

      return null;
    }
  })
};

```
So how do i define resolver for Date type using nestjs annotations  ? 
I've tried this but it doesn't work
```
@Resolver('Date')
export class DateResolver {

  @DelegateProperty('Date')
  getDate() {
    return new GraphQLScalarType({
      name: 'Date',
      description: 'Date custom scalar type',
      parseValue(value) {
        return new Date(value); // value from the client
      },
      serialize(value) {
        return value.getTime(); // value sent to the client
      },
      parseLiteral(ast) {
        if (ast.kind === Kind.INT) {
          return parseInt(ast.value, 10);
        }

        return null;
      }
    });
  }
}
```

**Comment by partyka95** [comment]
@valburyakov I also had a problem with it.

Solution:

date-time.scalar-type.ts
```
// ...
export const GraphQLDateTime = new GraphQLScalarType({
  name: 'DateTime',
  parseValue(value) {
    return new Date(value);
  },
  serialize(value) {
    return moment(value).toISOString();
  },
  parseLiteral(ast) {
    if (ast.kind === Kind.STRING) {
      return new Date(ast.value);
    }

    return null;
  },
});
```

bootstrap.ts
```
/** Scalar types **/
import { scalarTypes } from './type-implementations';

// ....

export class AppModule {

  // .........

  private createSchema(): GraphQLSchema {
    const schema = this.graphQLFactory.createSchema({
      typeDefs: [
        this.graphQLFactory.mergeTypesByPaths('./**/*.graphqls'),
      ],
      resolvers: {
        ...scalarTypes,
      },
      directiveResolvers: {},
    });

    const delegates = this.graphQLFactory.createDelegates();

    return mergeSchemas({
      schemas: [
        schema,
      ],
      resolvers: delegates,
    });
  }
}
```

But resolvers in createSchema function have problem, because this function overwrite resolvers property. I created pull request to fix it. Maybe there are other better solutions, but I have not found one.

**Comment by valburyakov** [comment]
@partyka95 Thx for the solution. But for me it still doesn't work, I did like in your example but no resolver for custom scalar is loaded:

app.module.ts
```
createSchema() {
    const typeDefs = this.graphQLFactory.mergeTypesByPaths('./**/*.graphql');
    const schema = this.graphQLFactory.createSchema({
      typeDefs,
      resolvers: {
        Date: GraphQLDateTime
      },
      directiveResolvers: {},
    });

    const delegates = this.graphQLFactory.createDelegates();

    return mergeSchemas({
      schemas: [
        schema,
      ],
      resolvers: delegates
    });
  }
```
Or this solution is working only with your fix in PR ? And show your `scalarTypes `.

**Comment by partyka95** [comment]
@valburyakov :

https://github.com/excitement-engineer/graphql-iso-date

https://github.com/nestjs/graphql/pull/8

**Comment by dcasier** [comment]
Another temporary solution :

` 
  import { ResolversExplorerService } from '@nestjs/graphql/resolvers-explorer.service'
  import { makeExecutableSchema } from 'graphql-tools'
  
  (...)

  createSchema() {
    const typeDefs = this.graphQLFactory.mergeTypesByPaths('./**/*.gql')
    //const schema = this.graphQLFactory.createSchema(typeDefs)

    const resolvers = this.resolversExplorerService.explore()
    Object.assign(resolvers, resolverFunctions)

    const schema = makeExecutableSchema({
      typeDefs,
      resolvers: resolvers,
    })
    return schema
  }`

**Comment by kamilmysliwiec** [comment]
Fixed thanks to #8 pull request. I'll push it together with the incoming release 🙂
